include/os/linux/spl/sys/vmem.h

```diff
@@ -96,11 +96,15 @@ extern size_t vmem_size(vmem_t *vmp, int typemask);
 
 #define	vmem_alloc(sz, fl)	spl_vmem_alloc((sz), (fl), __func__, __LINE__)
 #define	vmem_zalloc(sz, fl)	spl_vmem_zalloc((sz), (fl), __func__, __LINE__)
+#define	kvmem_alloc(sz, fl)	spl_kvmem_alloc((sz), (fl), __func__, __LINE__)
+#define	kvmem_zalloc(sz, fl)	spl_kvmem_zalloc((sz), (fl), __func__, __LINE__)
 #define	vmem_free(ptr, sz)	spl_vmem_free((ptr), (sz))
 #define	vmem_qcache_reap(ptr)	((void)0)
 
 extern void *spl_vmem_alloc(size_t sz, int fl, const char *func, int line);
 extern void *spl_vmem_zalloc(size_t sz, int fl, const char *func, int line);
+extern void *spl_kvmem_alloc(size_t sz, int fl, const char *func, int line);
+extern void *spl_kvmem_zalloc(size_t sz, int fl, const char *func, int line);
```

Comment:
Are these changes still needed or can be cleaned up?

Reply:
right now. its not required anymore. of course i still want to use kvmem in the mempool allocator. just centos prevents it right now due the weired behaviour. for me this is a nonstandard linux. now consider that kvmalloc was introduced in 4.4 and in later versions like 4.14 it turned to be heavily used in the linux kernel itself. so we can still use it if we make it depenend from the kernel version used. centos7 is 3.10 because if centos will use later kernels it will break itself if they dont fix the issue

Reply:
Amazon is 4.19
You could make it Kernel => 4.20 ?

Reply:
Full ACK, but comment it accordingly